src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/TripPatternForDate.java

```diff
@@ -98,6 +102,28 @@ public TripPatternForDate(
           .asDateTime(localDate, last.getArrivalTime(last.getNumStops() - 1))
           .toLocalDate();
     }
+    if (startOfRunningPeriod.isAfter(endOfRunningPeriod)) {
+      if (hasFrequencies()) {
+        var tripId = frequencies.get(0).tripTimes.getTrip().getId();
+        LOG.warn(
+          "Could not construct as start of the running period {} is after the end {} in frequency trip {}",
+          startOfRunningPeriod,
+          endOfRunningPeriod,
+          tripId
+        );
+      } else {
+        var tripId = tripTimes.get(0).getTrip().getId();
+        LOG.warn(
+          "Could not construct as start of the running period {} is after the end {} in trip {}",
+          startOfRunningPeriod,
+          endOfRunningPeriod,
+          tripId
+        );
+      }
```

Comment:
We don't know which of the trips is offending, so we can just log the pattern. We could also check that all trips are within the running period, and log the ones which are not.

Reply:
We know that there is a problem with either the first or the last trip. I changed this now so it logs the id of the last trip as well, I think it's enough that we got those two ids.

Reply:
Yes, but it will still cause issues later on, if the trips are not sorted as expected.

Reply:
Are the trips actually used anywhere if this this trip pattern for date gets ignored?

Reply:
No, but this will not be ignored, if the erroneous trip is not the las or first one

Reply:
Ok that is a good point. We would need to always check that none of the trip times are outside of the period or that they are all in order? Would it matter from performance perspective here?